src/validation.cpp

```diff
@@ -746,7 +746,10 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
                 // Applications relying on first-seen mempool behavior should
                 // check all unconfirmed ancestors; otherwise an opt-in ancestor
                 // might be replaced, causing removal of this descendant.
-                if (!SignalsOptInRBF(*ptxConflicting)) {
+                //
+                // If replaceability signaling is ignored due to node setting,
+                // replacement is always allowed.
+                if (!SignalsOptInRBF(*ptxConflicting) && !gArgs.GetBoolArg("-fullrbf", DEFAULT_TRANSACTION_REPLACEMENT)) {
```

Comment:
I would prefer parsing this setting once in the initialization, then passing it in to validation, instead of caling `GetBoolArg` for every transaction validation.

Reply:
+1, i think it should be a member of `ATMPArgs`.

Reply:
Agree, definitely shouldn't be querying gArgs here. Could also be passed in to the `MemPoolAccept` constructor since it wouldn't change across different submissions.

>  I'm thinking to introduce a new PolicySettings capturing fullrbf and other things like maxmempool, mempoolminfee, etc. It would be parsed at init with a InitPolicySettings() and given to NodeContext that way making accessible to the rpcs

sounds like a great idea 🚀

Reply:
Feel free to reference #25290 for how I do the options parsing/passing. Although we should resolve this before the PR is merged, it is an implementation detail and can be addressed later in the review cycle if there are more important things to sort out first.

Reply:
The `MemPoolOptions` wheel made there sounds really good to me so I went to rebase this PR on top of #25290 and just added a `m_full_rbf`.

Conceptually, I think it's even better to have the policy options in the mempool struct rather than `ATMPArgs` as the settings expected to be static across the run of a bitcoind instance.